Golem.Tests/JobStatusTests.cs

```diff
@@ -341,5 +343,162 @@ public async Task RequestorBreaksAgreement_KillingYagna()
+            Job? currentJob = await ReadChannel<Job?>(jobChannel);
+            Assert.NotNull(currentJob);
+
+            var jobStatusChannel = JobStatusChannel(currentJob);
+            var agreementId = currentJob.Id;
+
+            // Wait until ExeUnit will be created.
+            // Workaround for situations, when status update was so fast, that we were not able to create
+            // channel yet, so waiting for update would be pointless.
+            if (currentJob.Status != JobStatus.Computing)
+                Assert.Equal(JobStatus.Computing, await ReadChannel(jobStatusChannel,
+                    (JobStatus s) => s == JobStatus.DownloadingModel || s == JobStatus.Idle));
+            // Let him compute for a while.
+            await Task.Delay(2 * 1000);
+
+            _logger.LogInformation("=================== Killing Provider Agent ===================");
+            // Provider is killed so he is not able to terminate Agreement.
+            // Yagna has chance to be closed gracefully and close net connection with Requestor.
+            // If we would double Stop here, re-establishing connection would last longer.
+            var golemStatusChannel = StatusChannel(golem);
+            var pid = golem.GetProviderPid();
+            if (pid.HasValue)
+                Process.GetProcessById(pid.Value).Kill(true);
+
+            _logger.LogInformation("=================== Killing App ===================");
+            // Note: Provider needs to send termination with correct Reason.
+            // It's better to avoid yapapi finding out that Provider stopped working, because it
+            // could take action otherwise.
+            await app.Stop(StopMethod.SigKill);
+
+            await AwaitValue<JobStatus>(jobStatusChannel, JobStatus.Interrupted, TimeSpan.FromSeconds(30));
+            Assert.Equal(JobStatus.Interrupted, currentJob.Status);
+            await AwaitValue<Job?>(jobChannel, null, TimeSpan.FromSeconds(1));
+            Assert.Null(golem.CurrentJob);
+
+            await AwaitValue(golemStatusChannel, GolemStatus.Error, TimeSpan.FromSeconds(3));
+
+            _logger.LogInformation("=================== Restart Provider Agent ===================");
+            await golem.Start();
+
+            Assert.Null(golem.CurrentJob);
+
+            // ListJobs is one of things that can trigger termination of interrupted job.
+            // This will happen as well, when new event from yagna API will be processed.
+            var jobs = await golem.ListJobs(DateTime.Now - TimeSpan.FromMinutes(1));
+            var prevJob = jobs.Find(job => job.Id == agreementId) as Job;
+
+            Assert.NotNull(prevJob);
+            Assert.Equal(JobStatus.Interrupted, prevJob.Status);
+
+            _logger.LogInformation("=================== Requestor terminates Agreement ===================");
+            // Try to terminate Agreement with Reason which in normal situation would be treated as correct termination
```

Comment:
Do I understand correctly that the current situation isn't "normal" because this time one side already noticed Agreement termination (due to an interruption and subsequent ListJobs call)?
And that the latter two lines of the comment describe the behavior if the systemd hadn't noticed the Agreement termination, right (the sentence is a bit unclear to me)?

Reply:
Yes, you understand correctly. I will try to improve description